subsys/testsuite/coverage/coverage.c

```diff
@@ -23,7 +23,7 @@
 K_HEAP_DEFINE(gcov_heap, MALLOC_MAX_HEAP_SIZE);
 
 
-static struct gcov_info *gcov_info_head;
+static struct gcov_info *gcov_info_head = NULL;
```

Comment:
This is not necessary: the pointer will be default initialized to zero by the C runtime startup.